scipy/stats/_continuous_distns.py

```diff
@@ -6121,19 +6122,82 @@ def _entropy(self, c):
     def fit(self, data, *args, **kwds):
         parameters = _check_fit_input_parameters(self, data, args, kwds)
         data, fshape, floc, fscale = parameters
-        if floc is None:
-            return super().fit(data, **kwds)
+
+        ndata = data.shape[0]
+
+        def get_shape(scale, location):
+            # The partial derivative w.r.t. the shape can be solved in
+            # closed form for the shape.
+            return ndata / np.sum(np.log((data - location) / scale))
+
+        if floc is fscale is None:
+            # Optimization over a system of equations is possible when the
+            # location of the data is not set. The optimization requires that
+            # the scale be free, but the shape may be fixed. The equations
+            # are from the partial derivatives of the log-likelihood function.
+
+            def dLdScale(shape, scale):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the scale.
+                return ndata * shape / scale
+
+            def dLdLocation(shape, location):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the location.
+                return (shape + 1) * np.sum(1 / (data - location))
+
+            def fun_to_solve(scale):
+                # optimize over the scale, setting the partial derivatives
+                # w.r.t. to location and scale equal.
+                location = np.min(data) - scale
+                shape = fshape or get_shape(scale, location)
+                return dLdLocation(shape, location) - dLdScale(shape, scale)
+
+            def interval_contains_root(lbrack, rbrack):
+                # return true if the signs disagree.
+                return (np.sign(fun_to_solve(lbrack)) !=
+                        np.sign(fun_to_solve(rbrack)))
+
+            # set brackets for `root_scalar` to use when optimizing over the
+            # scale such that a root is likely between them.
+            lbrack, rbrack = .1, np.max(data)
+            # if a root is not between the brackets, iteratively expand them
+            # until they include a sign change, checking after each bracket is
+            # modified.
+            while not interval_contains_root(lbrack, rbrack):
+                lbrack /= 2
+                rbrack *= 2
```

Comment:
This looks much better. Probably should do something to avoid infinite loop. The simplest thing would be:
```suggestion
            while (not interval_contains_root(lbrack, rbrack) 
                   and (lbrack > 0 or rbrack < np.inf)):
                lbrack /= 2
                rbrack *= 2
```
Then `root_scalar` would error out. I don't expect this to happen, so I don't think we need to have a backup (e.g. resort to `super.fit`). Just a precaution to avoid infinite loop. _Update: happily, an infinite bracket leads to `res.converged=False` and then `super.fit` runs anyway. That's fine._